File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        
        StringBuilder res = new StringBuilder();
        
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            
            if(res.length() > 0 && res.charAt(res.length()-1) == c){
                res.setLength(res.length()-1);
            }else{
                res.append(c);
            }
            
        }
        
        return res.toString();
    }
}

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        st = []
        for i in s: # a b b a c
            if len(st)>0: 
                el = st[-1] # a b a 
                if el == i: # F T 
                    st.pop() 
                    continue
            st.append(i) # a  ab
        return "".join(st)
        